### DISCUSSION TITLE
Can I create a new database when it does not exist by sqlalchemy?

### DISCUSSION THREAD

**Discussion by monchin** [sqlalchemy/sqlalchemy#11695] [Usage Questions] [answered]
I'm using `postgresql+pg8000` as engine, and my database uri includes my username and password, and a database that doesn't exist. When I try to create engine, it says
```
sqlalchemy.exc.ProgrammingError: (pg8000.dbapi.ProgrammingError) {'S': 'FATAL', 'V': 'FATAL', 'C': '3D000', 'M': 'database "xxx" does not exist', 'F': 'postinit.c', 'L': '885', 'R': 'InitPostgres'}
```
But if I change database to an already existed one, everything works well.

I wonder if I can use sqlalchemy to check whether the database exists (by `try except`), and if the database dosen't exist, I can use sqlalchemy to create that database? I tried to drop database name from uri, create engine from that name-dropped uri, and use `engine.execute("CREATE DATABASE xxx")`, but it would recognize my username as a database name and raise error in `create_engine`. Must I create new database manually?

**Comment by CaselIT** [comment, marked as answer]
Hi,

Every connection in postgresql needs to connect to an existing database, so trying to connect to a not existing data will error in any tool.

An usual solution is to connect to template1, check of the db exist, create it if not, then connect to it.

**Reply by monchin** [reply]
Thank you! I'll try it!

**Reply by monchin** [reply]
I found in [this answer](https://stackoverflow.com/questions/6506578/how-to-create-a-new-database-using-sqlalchemy/30971098#30971098) that we can use `sqlalchemy_utils` to do that, it is a relatively easy way.

**Reply by CaselIT** [reply]
Sure, but it's not overly hard to do without that. Sqlalchemy utils does as I mentioned above
